src/PKSim.Core/Commands/ResetParameterCommand.cs

```diff
@@ -22,7 +24,25 @@ protected override void ExecuteUpdateParameter(IExecutionContext context)
 
       protected override void UpdateParameter(IParameter parameter, IExecutionContext context)
       {
-         parameter?.ResetToDefault();
+         if (parameter == null) return;
+         parameter.ResetToDefault();
+         SaveValueOriginFor(parameter);
+         resetValueOriginForDefaultParameter(parameter, context);
+      }
+
+      private void resetValueOriginForDefaultParameter(IParameter parameter, IExecutionContext context)
+      {
+         var parametersInContainerRepository = context.Resolve<IParametersInContainerRepository>();
```

Comment:
In case of a reset command,  I did implement the concept of retrieving the default value origin if available. The update is only performed for default parameter